scrapper.py

```diff
@@ -31,67 +53,226 @@ class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls: list, max_articles: int):
-        pass
+    seed_urls: List[str]
+    max_articles: int
+    max_articles_per_seed: int
+    urls: Set[str]
+
+    def __init__(self, seed_urls: list, max_articles: int, max_articles_per_seed: int):
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.max_articles_per_seed = max_articles_per_seed
+        self.urls = set()
+
+    @staticmethod
+    def _extract_url(article_bs: BeautifulSoup) -> List[str]:
+        """
+        Extracts news urls from the seed page
+        """
+        news = article_bs.select('div.col-md-9 div.col-md-6 h2 a')
+        links = [article.attrs['href'] for article in news]
+
+        return links
+
+    @staticmethod
+    def save_interim(**args) -> None:
```

Comment:
Crawler should not keep the state. it could be the role of CrawlerRecursive

Reply:
addressed this issue